src/plugins/tables/tables.js

```diff
@@ -253,12 +254,12 @@ $document.on( "submit", ".wb-tables-filter", function( event ) {
 		if ( $elm.is( "select" ) ) {
 			$value = $elm.find( "option:selected" ).val();
 		} else if ( $elm.is( "input[type='number']" ) ) {
-			var $val = $elm.val(), $minNum, $maxNum;
+			var $minNum, $maxNum;
 
 			// Retain minimum number (always the first number input)
 			if ( $cachedVal === "" ) {
 				$cachedVal = parseFloat( $val );
-				$cachedVal = ( $cachedVal ) ? $cachedVal : "-0";
+				$cachedVal = ( $cachedVal ? $cachedVal : $val );
```

Comment:
Humm, we will need to better understand the use of "-0" here and how it used later. May be we can look at the history to find out more information about the use case of using the value "-0"
 
If the value of in the input don't parse, then we might just fallback on a simple text search matching the input value.

Reply:
I originally coded a lot of these plugins. Quite a few of these changes are completely unrelated to the date filter and are related the the number filter.

Does the original poster have example data that can be used to show when the date filter breaks. If so I can see if I’m able to run some test.

Reply:
Yes for the date filter function yes I can see that issue. But the code you are changing on this line is unrelated to the date filter.